lib/src/extensions.dart

```diff
@@ -204,6 +204,24 @@ extension DateTimeTimeExtension on DateTime {
             microsecond ?? this.microsecond,
           );
   }
+
+  /// Returns the Monday of this week
+  DateTime get firstDayOfWeek => DateTime.utc(year, month, day + 1 - weekday);
```

Comment:
I would like us to check if the current `DateTime` `isUtc` before constructing a `utc` variant. 
For example, see line 185